src/transformers/models/pvt_v2/configuration_pvt_v2.py

```diff
@@ -0,0 +1,165 @@
+            many stages the model has). If unset and `out_features` is set, will default to the corresponding stages.
+            If unset and `out_features` is unset, will default to the last stage.
+    Example:
+
+    ```python
+    >>> from transformers import PvtV2Model, PvtV2Config
+
+    >>> # Initializing a pvt_v2_b0 style configuration
+    >>> configuration = PvtV2Config()
+
+    >>> # Initializing a model from the OpenGVLab/pvt_v2_b0 style configuration
+    >>> model = PvtV2Model(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "pvt_v2"
+
+    def __init__(
+        self,
+        image_size: Union[int, Tuple[int, int]] = 224,
+        num_channels: int = 3,
+        num_encoder_blocks: int = 4,
+        depths: List[int] = [2, 2, 2, 2],
+        sr_ratios: List[int] = [8, 4, 2, 1],
+        hidden_sizes: List[int] = [32, 64, 160, 256],
+        patch_sizes: List[int] = [7, 3, 3, 3],
+        strides: List[int] = [4, 2, 2, 2],
+        num_attention_heads: List[int] = [1, 2, 5, 8],
+        mlp_ratios: List[int] = [8, 8, 4, 4],
+        hidden_act: Union[str, Callable] = "gelu",
+        hidden_dropout_prob: float = 0.0,
+        attention_probs_dropout_prob: float = 0.0,
+        initializer_range: float = 0.02,
+        drop_path_rate: float = 0.0,
+        layer_norm_eps: float = 1e-6,
+        qkv_bias: bool = True,
+        linear_attention: bool = False,
+        out_features=None,
+        out_indices=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        image_size = (image_size, image_size) if isinstance(image_size, int) else image_size
+
+        self.image_size = image_size
+        self.num_channels = num_channels
+        self.num_encoder_blocks = num_encoder_blocks
+        self.depths = depths
+        self.sr_ratios = sr_ratios
+        self.hidden_sizes = hidden_sizes
+        self.patch_sizes = patch_sizes
+        self.strides = strides
+        self.mlp_ratios = mlp_ratios
+        self.num_attention_heads = num_attention_heads
+        self.hidden_act = hidden_act
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.attention_probs_dropout_prob = attention_probs_dropout_prob
+        self.initializer_range = initializer_range
+        self.drop_path_rate = drop_path_rate
+        self.layer_norm_eps = layer_norm_eps
+        self.qkv_bias = qkv_bias
+        self.linear_attention = linear_attention
+        self.stage_names = [f"stage{idx}" for idx in range(1, len(depths) + 1)]
+        self._out_features, self._out_indices = get_aligned_output_features_output_indices(
+            out_features=out_features if out_features is not None else getattr(self, "out_features", None),
+            out_indices=out_indices if out_indices is not None else getattr(self, "out_indices", None),
+            stage_names=self.stage_names,
+        )
```

Comment:
I don't see why we need this here - neither `out_features` not `out_indices` should be already set when initializing the config 

```suggestion
        self._out_features, self._out_indices = get_aligned_output_features_output_indices(
            out_features=out_features,
            out_indices=out_indices,
            stage_names=self.stage_names,
        )
```

Reply:
Ahhh, ok I see what happened. This was a tricky one. The order of the base classes matters here, as before we had `class ResNetConfig(BackboneConfigMixin, PretrainedConfig)` but `class PvtV2Config(PretrainedConfig, BackboneConfigMixin)`, and thus the latter was not getting the proper `to_dict` method override, and therefore not saving these fields correctly to JSON, which led to the weirdness during the `__init__` call

I have fixed this the correct way now in: https://github.com/huggingface/transformers/pull/26812/commits/0e7a0545296546f33fc3e9e1bbf0f2e78ae43e20

Thank you for your scrutiny here, I hadn't realized these values were not being saved properly. For a bit of backstory, I built the PvtV2 and was using it *before* I incorporated the Backbone mixins, so the code you highlighted was a hack that was part of the original work and prevented me from seeing this issue.

Reply:
Thanks for digging into this. Adding on my to-do list to make sure this is easier to implement / catch